### PR TITLE
Milestone 1 : Battleship

### PR DESCRIPTION
**Pull request by tiagoit**
### First milestone
Vanilla JS implementation of the classic Battleship board game.
The goal of this project is practicing the usage of:

  - ES6 Factory functions
  - ES6 Modules
  - Webpack
  - TDD with Jest

### Features added to this milestone:
- Project setup with Webpack and jest.
- Controllers for Game, Board, Player.
- Views for Game and Board.
- Models for Player and Ship.

### Links
- **Assignment page** - [The Odin Project](https://www.theodinproject.com/courses/javascript/lessons/battleship)

### CODE REVIEW

File: src/controllers/game-board.controller.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import GameBoardView from '../views/game-board.view';
+import AppService from '../app.service';
+
+const GameBoardController = (pType) => {
+  const boardArray = []; // [0|1|Ship]
+  const ships = [];
+  const playerType = pType; // [AI|Human]
+  const appService = AppService();
+
+  // Init a 10x10 board filled with zeros.
```
Comment (bellom) [inline src/controllers/game-board.controller.js:11]: **Kindly remove commented code inside this file, you can always use git history to access previous code right**:slightly_smiling_face:

File: src/controllers/game-board.controller.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        return true;
+      }
+    }
+
+    // NO ship | HAVE shot (cell === 1)
+    // HAVE ship | HAVE shot cell.hits[inShipIndex] === 1
+    return false;
+  };
+
+  // Returns true if all the ships on the board have sunk.
```
Comment (bellom) [inline src/controllers/game-board.controller.js:81]: **Kindly remove commented code inside file, you can always use git history to access previous code right**:slightly_smiling_face:

File: src/controllers/game.controller.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import GameView from '../views/game.view';
+import PlayerController from './player.controller';
+import AppService from '../app.service';
+
+const GameController = () => {
+  const players = [];
+  const appService = AppService();
+  let gameInProgress = false;
+
+  // Game is finished if any board have all ships sinked.
```
Comment (bellom) [inline src/controllers/game.controller.js:10]: **Kindly remove commented code inside file, you can always use git history to access previous code right**:slightly_smiling_face:

File: src/controllers/player.controller.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const PlayerController = (n) => {
+  const board = GameBoardController(n === 'AI' ? 'AI' : 'Human');
+  const appService = AppService();
+  const name = n;
+
+  const aiPlaceShips = () => {
+    // TODO: Remove
+    board.placeShip(0, 0, 3, false);
+
+    // let shipBlocks = 20;
```
Comment (bellom) [inline src/controllers/player.controller.js]: **Kindly remove commented code inside file, you can always use git history to access previous code right**:slightly_smiling_face:

File: src/views/game-board.view.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    if (playerType === 'AI') {
+      const boardEl = document.getElementsByTagName('ai-board')[0];
+      boardEl.innerHTML = boardHtml;
+      boardEl.querySelectorAll('.b-row > div').forEach((el) => {
+        el.addEventListener('click', (ev) => {
+          const ds = ev.target.dataset.id ? ev.target.dataset : ev.target.parentElement.dataset;
+          const x = parseInt(ds.id[0], 10);
+          const y = parseInt(ds.id[1], 10);
+          callback({ x, y });
+          // this.battle(boardArray, playerType, callback);
```
Comment (bellom) [inline src/views/game-board.view.js]: **Kindly remove commented code inside file, you can always use git history to access previous code right**:slightly_smiling_face:

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 
#### Awesome job so far!! :+1: 
#### You have done wonderfully well, test works all fine, you passed `almost` all the checklist for this project :+1:, just few issues left to work on............ :slightly_smiling_face: 

Below are part of the checklist for this milestone, kindly work on these few things listed below;
**1. About Pull Request title and short summary:** it shouldn't look like `README` file, you can easily just add the features added to the project, example below;
image
Example; 
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/

**2. kindly check out the comments below**

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: APPROVED

Hi Team, good job on this one :clap:, it works fine and I see changes from the previous reviewer has been made, just as a side note, I feel there a way too much comments in the code, a lot of the comments seems not so needed, please try to favour clearer varaible and method names over comments. 
**This is approved**

Raphael Noriode :sunglasses:

**Pull request merged**